File: java/BubbleSort.java
public class BubbleSort {
     public static void main(String[] args) {
         int[] tab = {7, 2, 10, 3, 5, 8};
 
         System.out.println("Tablica przed sortowaniem: ");
         printArray(tab);
 
         bubbleSort(tab);
 
         System.out.println("Tablica po sortowaniu: ");
         printArray(tab);
     }
 
     public static void bubbleSort(int[] tab) {
         int n = tab.length;
         boolean swapped;
 
         do {
             swapped = false;
             for (int i = 1; i < n; i++) {
                 if (tab[i - 1] > tab[i]) {
                     int temp = tab[i - 1];
                     tab[i - 1] = tab[i];
                     tab[i] = temp;
                     swapped = true;
                 }
             }
             n--;
         } while (swapped);
     }
 
     public static void printArray(int[] tab) {
         for (int num : tab) {
             System.out.print(num + " ");
         }
         System.out.println();
     }
 }

File: kotlin/BubbleSort.kt
fun main() {
    val tab = arrayOf(7, 2, 10, 3, 5, 8)

    println("Tablica przed sortowaniem:")
    printArray(tab)

    bubbleSort(tab)

    println("Tabica po sortowaniu:")
    printArray(tab)
}

fun bubbleSort(tab: Array<Int>) {
    var n = tab.size
    var swapped: Boolean

    do {
        swapped = false
        for (i in 1 until n) {
            if (tab[i - 1] > tab[i]) {
                val temp = tab[i - 1]
                tab[i - 1] = tab[i]
                tab[i] = temp
                swapped = true
            }
        }
        n--
    } while (swapped)
}

fun printArray(tab: Array<Int>) {
    for (num in tab) {
        print("$num ")
    }
    println()
}